core/src/util.jl

```diff
@@ -750,3 +750,27 @@ function get_basin_idx(edge_metadata::EdgeMetadata, id::NodeID)::Int32
         0
     end
 end
+
+function set_initial_allocation_mean_flows!(integrator)::Nothing
+    (; u, p, t) = integrator
+    (; allocation, graph, basin) = p
+    (; mean_flows, allocation_models) = allocation
+    (; Δt_allocation) = allocation_models[1]
```

Comment:
Does this work if `use_allocation = true` but there are no allocation models?

Reply:
I added a check before the call of this function to prevent that situation